### PR TITLE
minetest rev2 & fix irrlicht 1.9.0mt8

### PR DESCRIPTION
**Pull request by Zweihorn**
PR ready for REVIEW

#### Description

Combined **amend** minetest rev2 & **fix** irrlicht 1.9.0mt8 rev1

- **Amend Profile** to MT 5.6.1 "minetest" rev2
  + **New variants:** [+]GLES, benchmark, debug, gprof, noclient, psql, server
  + **Fixes** irrlichtmt GLES & minetest - ref https://trac.macports.org/ticket/66599
- **Fixes** to irrlicht 1.9.0mt8 rev1
  + **Fix01:** irrlichtmt fails to install on Mac OS 10.6 - ref <https://trac.macports.org/ticket/66439>
  + **Fix02:** minetest arm64 support - ref <https://trac.macports.org/ticket/66600>

ref https://github.com/minetest/irrlicht/issues/152

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 11.7.2 20G1020 x86_64
Command Line Tools 13.2.0.0.1.1638488800

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

###### Future tasks:

The b.m. Trac ticket should **not hinder** any review and is **an independent activity** by same maintainer:
- ref https://trac.macports.org/ticket/66610

### CODE REVIEW

**Comment by Zweihorn** [comment]
ERROR - This is somewhat unexpected. 

 ```
. . .
  -- Detecting CXX compile features - done
  -- *** Will build version 5.6.1-MacPorts-rev2 ***
  -- Found IrrlichtMt 1.9.0.9
  CMake Error at CMakeLists.txt:135 (message):
    IrrlichtMt 1.9.0mt8 is required to build
  
  
  -- Configuring incomplete, errors occurred!
  ```

AFAIK the a.m. test build should have combined IrrlichtMt 1.9.0.9 with MT 5.6.1-MacPorts-rev2
However, I have to check again as the port install of MT could have just grabbed the 1.9.0mt8 and thus _ignored_ the new IrrlichtMt 1.9.0.9 build apparently. Anyway not a big problem and I will introduce the a.m. fixes to 1.9.0mt8 as backport.

A good learning experience again.

@l2dy - Sorry for the newbie noise produced by me.

Back to the drawing board ...

**Comment by Zweihorn** [comment]
**Ready for review...**

And the skeletons in the drawer are:
- https://github.com/minetest/irrlicht/issues/152
- #17153 
  - https://github.com/macports/macports-ports/pull/17153#issuecomment-1374599091
- #17161

File: devel/irrlichtmt/Portfile [hunk 1]
```diff
@@ -19,3 +20,29 @@ checksums           rmd160  4da410c24f67e8c0ebd9880308b8d796d2f86676 \
 depends_lib         path:include/turbojpeg.h:libjpeg-turbo \
                     port:libpng \
                     port:zlib
+
+pre-build {
```
Comment (aeiouaeiouaeiouaeiouaeiouaeiou) [inline devel/irrlichtmt/Portfile:24]: [`mesa`](https://ports.macports.org/port/mesa/) already includes both headers. It maybe better to copy a fragment from Minetest:

`depends_build    port:mesa`

File: games/minetest/Portfile [hunk 2]
```diff
@@ -109,8 +112,104 @@ configure.args-append   -DCMAKE_BUILD_TYPE=Release \
                         -DENABLE_LUAJIT=ON \
                         -DLUA_INCLUDE_DIR:PATH=${prefix}/include/luajit-2.1 \
                         -DLUA_LIBRARY:FILEPATH=${prefix}/lib/libluajit-5.1.dylib \
+                        -USE_GPROF=FALSE \
                         -DVERSION_EXTRA=MacPorts-rev${revision}
 
+default_variants        +GLES
```
Comment (aeiouaeiouaeiouaeiouaeiouaeiou) [inline games/minetest/Portfile:118]: Can't we make OpenGL ES be used only on Snow Leopard (as it is unnecessary for newer systems)? I had the same idea to do this, but never got around to testing it on a virtual machine.

File: devel/irrlichtmt/Portfile (hunk 1 above)
Comment (Zweihorn) [reply devel/irrlichtmt/Portfile:24]: THX

Please be aware there was an argument that I always had to heed the work of my (unknown) predecessor as maintainer and that to always include "all the paths" to me as the new maintainer. during rev1.
AFAIK see @reneeotten in #17080 if I reall this recent discussion correctly.

Furthermore, this "mess" basically started with my innocent idea of dropping "mesa" and GLES completely by kind advice from @l2dy and discussion at:
- https://trac.macports.org/ticket/66599

@aeiouaeiouaeiouaeiouaeiouaeiou  To better disambiguate: By a.m. "both headers" you mean the top two of three headers i.e. "libjpeg-turbo" and "libpng" I presume?

Comment (Zweihorn) [reply devel/irrlichtmt/Portfile:24]: Might drop both headers i.e. "libjpeg-turbo" and "libpng" in due time.

File: games/minetest/Portfile (hunk 2 above)
Comment (Zweihorn) [reply games/minetest/Portfile:118]: Again at https://trac.macports.org/ticket/66599

Furthermore, please be aware that upstream MT devs apparently introduced more GLES statements into new code to **upstream "irrlicht 1.9.0mt9"** only recently by: https://github.com/minetest/irrlicht/releases/tag/1.9.0mt9

Hope this helps.

Comment (Zweihorn) [reply games/minetest/Portfile:118]: IMO there is nothing wrong with '+GLES' and one could make use of "minetest -GLES" anyway. Only proofs to me that my idea of presenting GLES by variant was a good idea.

GLES stays until further notice.

File: devel/irrlichtmt/Portfile (hunk 1 above)
Comment (Zweihorn) [reply devel/irrlichtmt/Portfile:24]: Anyway, why should I drop both headers i.e. 'libjpeg-turbo' and 'libpng' from "irrlichtmt" port?

@l2dy is the "irrlichtmt" maintainer and I am the just the new maintainer to "minetest" providing some fixes after discussion with same "irrlichtmt" maintainer.

Please note I am **quite reluctant to gain too much influence** on other people's ports and I will always appreciate their efforts as maintainers of their port and distribution to FOSS.

Comment (aeiouaeiouaeiouaeiouaeiouaeiou) [reply devel/irrlichtmt/Portfile:24]: >@aeiouaeiouaeiouaeiouaeiouaeiou To better disambiguate: By a.m. "both headers" you mean the top two of three headers i.e. "libjpeg-turbo" and "libpng" I presume?

I meant the header files - glext.h and khrplatform.h, which are attached to the commit.

**Comment by Zweihorn** [comment]
@aeiouaeiouaeiouaeiouaeiouaeiou 

Pleeeeeeeeeaaaaaaase kindly consider approval of review on basis of my a.m. resolve. 

Anyway there will be another day with "minetest" rev3 in the future, I presume.

**Review by aeiouaeiouaeiouaeiouaeiouaeiou — APPROVED** [review]

File: devel/irrlichtmt/Portfile (hunk 1 above)
Comment (Zweihorn) [reply devel/irrlichtmt/Portfile:24]: > > @aeiouaeiouaeiouaeiouaeiouaeiou To better disambiguate: By a.m. "both headers" you mean the top two of three headers i.e. "libjpeg-turbo" and "libpng" I presume?
> 
> I meant the header files - glext.h and khrplatform.h, which are attached to the commit.

THX

Please have a look on the conditionals in a.m. `pre-built` script ans you could be aware:
- 'glext.h' is only applied to l**egacy-os** platforms Snow Leopard and maybe Leopard (afaik the latter fails due to other reasons unfortunately)
- 'khrplatform.h' applies only to **arm64** platforms for obvious reasons.

Comment (Zweihorn) [reply devel/irrlichtmt/Portfile:24]: @aeiouaeiouaeiouaeiouaeiouaeiou THX and you have a good heart.

**Comment by l2dy** [comment]
You should split this commit into 2, following the [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages). Here are some git commands for your reference, with `xxxx` replaced by the actual commit message.

```sh
# in repository root with a clean working tree
git reset --mixed HEAD~2
git commit -m 'irrlichtmt: xxxx' -- devel/irrlichtmt
git commit -m 'minetest: xxxx' -- games/minetest
git push -f
```

**Comment by Zweihorn** [comment]
@l2dy - Sorry, but let me ask what was your message in "combining and commit on top" before?

> You should split this commit into 2, following the [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages). Here are some git commands for your reference, with `xxxx` replaced by the actual commit message.
> 
> ```shell
> # in repository root with a clean working tree
> git reset --mixed HEAD~1
> git commit -m 'irrlichtmt: xxxx' -- devel/irrlichtmt
> git commit -m 'minetest: xxxx' -- games/minetest
> git push -f
> ```
> 
> Edit: 1 is the number of commits to move backwards in history, in your PR there is 1.

@l2dy - THX - May be the Port be with you.

Apparently I totally misunderstood and naturally I am to blame and my job of learning and improving as new maintainer. Don't want to be a bother to you, however, I sincerely would have preferred **some more clarity** in your before message.

Anyway, I already made it much more challenging to you and reviewers due to:
1. the valid expert's approach to avoid the variant mess;
2. the evident binding of MT 5.6 to IrrLichtMT8 and MT 5.7 to IrrlichtMT9

Resolving to a **new approach** with _several new ports & Portfiles_ for both 'ircchlichtmt' and 'minetest' respectively.

There will be fun.

🌻

**Comment by l2dy** [comment]
> @l2dy - Sorry, but let me ask what was your message in "combining and commit on top" before?

A PR could be made from a branch that has more than one commits to be merged. In most cases, it's recommended to isolate changes of different ports into different commits and stack them on top of each other in their order of dependency relationship.

If for any reason this is not feasible, you should still follow the "List any modified ports in Subject" rule and start your commit message with `irrlichtmt, minetest: `.

**Comment by Zweihorn** [comment]
### Old News

> If for any reason this is not feasible, you should still follow the "List any modified ports in Subject" rule and start your commit message with `irrlichtmt, minetest: `.

Good advice for some future use, I guess.

This PR was closed and the branch was deleted one hour ago.

### New Deal

On to a new approach with several new ports & Portfiles for both 'irrlichtmt' and 'minetest' respectively and indepent commits although a quite interdependent change to both ports will hopefully happen soon.

There will be fun.

🌻

**Pull request closed without merging**